bundles/org.openhab.binding.lifx/src/main/java/org/openhab/binding/lifx/internal/protocol/SetTileEffectRequest.java

```diff
@@ -0,0 +1,149 @@
+    private static final Field<Long> FIELD_DURATION = new UInt64Field().little();
+    private static final Field<Long> FIELD_RESERVED_19_TO_26 = new UInt32Field().little();
+    private static final Field<Long> FIELD_PARAMETER_27_TO_58 = new UInt32Field().little();
+    private static final Field<Integer> FIELD_PALLETTE_COUNT = new UInt8Field();
+    private static final Field<HSBK> FIELD_PALETTE_60_TO_187 = new HSBKField();
+
+    private final Logger logger = LoggerFactory.getLogger(SetTileEffectRequest.class);
+
+    private Integer reserved0 = 0;
+    private Integer reserved1 = 0;
+    private Long reserved19to22 = 0L;
+    private Long reserved23to26 = 0L;
+    private Effect effect;
+
+    public SetTileEffectRequest() {
+        setTagged(false);
+        setAddressable(true);
+        setAckRequired(true);
+    }
+
+    public SetTileEffectRequest(Effect effect) {
+        this();
+        this.effect = effect;
+    }
+
+    @Override
+    public int packetType() {
+        return TYPE;
+    }
+
+    @Override
+    protected int packetLength() {
+        return 188;
+    }
+
+    @Override
+    protected void parsePacket(ByteBuffer bytes) {
+        reserved0 = FIELD_RESERVED_0.value(bytes);
+        reserved1 = FIELD_RESERVED_1.value(bytes);
+        FIELD_INSTANCE_ID.value(bytes);
+        Integer effectType = FIELD_TYPE.value(bytes);
+        Long speed = FIELD_SPEED.value(bytes);
+        Long duration = FIELD_DURATION.value(bytes);
+        reserved19to22 = FIELD_RESERVED_19_TO_26.value(bytes);
+        reserved23to26 = FIELD_RESERVED_19_TO_26.value(bytes);
+        Long[] parameters = new Long[8];
+        for (int i = 0; i < parameters.length; i++) {
+            parameters[i] = FIELD_PARAMETER_27_TO_58.value(bytes);
+        }
+        Integer palletteCount = FIELD_PALLETTE_COUNT.value(bytes);
+        HSBK[] pallette = new HSBK[palletteCount];
+        for (int i = 0; i < pallette.length; i++) {
+            pallette[i] = FIELD_PALETTE_60_TO_187.value(bytes);
+        }
+        effect = new Effect(effectType, speed, duration, pallette);
+    }
+
+    @Override
+    protected ByteBuffer packetBytes() {
+        ByteBuffer buffer = ByteBuffer.allocate(packetLength());
+        buffer.put(FIELD_RESERVED_0.bytes(reserved0));
+        buffer.put(FIELD_RESERVED_1.bytes(reserved1));
+        buffer.put(FIELD_INSTANCE_ID.bytes(0L));
+        buffer.put(FIELD_TYPE.bytes(effect.getType()));
+        buffer.put(FIELD_SPEED.bytes(effect.getSpeed()));
+        buffer.put(FIELD_DURATION.bytes(effect.getDuration()));
+        buffer.put(FIELD_RESERVED_19_TO_26.bytes(reserved19to22));
+        buffer.put(FIELD_RESERVED_19_TO_26.bytes(reserved23to26));
+        for (int i = 0; i < 8; i++) {
+            buffer.put(FIELD_PARAMETER_27_TO_58.bytes(0L));
+        }
+        buffer.put(FIELD_PALLETTE_COUNT.bytes(effect.getPallette().length));
+        HSBK[] pallette = effect.getPallette();
```

Comment:
```suggestion
        HSBK[] palette = effect.getPalette();
```

Please also fix the other typo occurrences in this file.